File: go/scrabble-score/scrabble_score.go
package scrabble

import (
    "strings"
)
// LetterScore decides each letter worth.
func LetterScore(letter rune) int {
    switch letter {
        case 'q', 'z': return 10
        case 'j', 'x': return 8
        case 'k': return 5
        case 'f', 'h', 'v', 'w', 'y': return 4
        case 'b', 'c', 'm', 'p': return 3
        case 'd', 'g': return 2
        default: return 1
    }
}
// Score converts the word to lowercase and calls LetterScore on each of its letters.
func Score(word string) int {
    r := 0
    for _, l := range strings.ToLower(word) {
        r += LetterScore(l)
    }
    return r
}

File: python/scrabble-score/scrabble_score.py
SCORES = {
    1: ["A", "E", "I", "O", "U", "L", "N", "R", "S", "T"],
    2: ["D", "G"],
    3: ["B", "C", "M", "P"],
    4: ["F", "H", "V", "W", "Y"],
    5: ["K"],
    8: ["J", "X"],
    10: ["Q", "Z"]
}


def score(word: str) -> int:
    return sum(
        [
            score for score, letters in SCORES.items()
            for letter in word.upper() if letter in letters
        ]
    )
